batch/batch/worker/worker.py

```diff
@@ -318,6 +322,11 @@ async def wrapper(f, *args, **kwargs):
     return wrapper
 
 
+async def pull_docker_image(pool: concurrent.futures.ThreadPoolExecutor, image_ref_str: str, auth: dict):
+    assert docker_sync
+    return await blocking_to_async(pool, docker_sync.images.pull, image_ref_str, auth_config=auth)
```

Comment:
I'm confused by this. Why do we wrap a synchronous docker client with blocking_to_async instead of using aiodocker?

Reply:
If this is unavoidable, I think a cleaner fix is to wrap the aiodocker instance and override image pull to do this.

Reply:
Jackie would know the exact details, but I believe it had to do with aiodocker not always properly handling credentials in a way that messed up the azure credentials, and this was chosen in favor of monkey-patching aiodocker in the interim before we move over to podman.